lib/npc-generation/breakGenderNorms.ts

```diff
@@ -0,0 +1,84 @@
+import { random } from '../src/random'
+import { clamp } from '../src/utils'
+import { Town } from '../town/_common'
+import { GenderName } from './raceTraits'
+import { NPC } from './_common'
+
+/**
+ * town.roll.equality,
+ * chance of breaking gender norm
+ */
+const genderEqualityLikelihood: [number, number, GenderName, GenderName][] = [
+  [139, 10, 'woman', 'man'],
+  [99, 10, 'woman', 'man'],
+  [90, 15, 'woman', 'man'],
+  [80, 20, 'woman', 'man'],
+  [70, 30, 'woman', 'man'],
+  [60, 40, 'woman', 'man'],
+  [55, 60, 'woman', 'man'],
+  [50, 100, 'man', 'woman'],
+  [45, 60, 'man', 'woman'],
+  [40, 50, 'man', 'woman'],
+  [30, 40, 'man', 'woman'],
+  [20, 30, 'man', 'woman'],
+  [10, 20, 'man', 'woman'],
+  [5, 15, 'man', 'woman'],
+  [-101, 10, 'man', 'woman']
+]
+
+export function breakGenderNorms (town: Town) {
+  clamp(town.roll.equality, 0, 100)
+
+  // MIGRATION: Not available outside src.
+  /* if (settings.ignoreGender) {
+    return true
+  } */
+
+  const temp = genderEqualityLikelihood.find(([threshold]) => {
+    return threshold <= town.roll.equality
+  })
+
+  const percentage = temp ? temp[1] : 50
+  return random(1, 100) < percentage
+}
+
+export function isDominantGender (town: Town, npc: NPC) {
+  const temp = genderEqualityLikelihood.find(([threshold]) => {
+    return threshold <= town.roll.equality
+  })
+  if (temp) {
+    return npc.gender === temp[2]
+  }
+  throw new Error('Could not find match.')
+}
+
+export function initSexistProfession (town: Town, npc: NPC) {
+  if (npc.profession && !npc.gender) {
+    if (breakGenderNorms(town) === false) {
+      const newGender = checkProfessionGender(town, npc)
+      if (newGender) {
+        npc.gender = newGender
+      }
+    } else {
+      npc.isBreakingGenderNorms = true
+    }
+  }
+  return npc
+}
+
+// FIXME: Not sure what's going on here?
+function checkProfessionGender (town: Town, npc: NPC) {
+  /* const profession = findProfession(town, npc)
+  const genders = genderEqualityLikelihood.find(([threshold]) => {
+    return threshold <= town.roll.equality
+  })
+  if (genders) {
+    switch (genders.indexOf(profession.domSub)) {
+      case 2:
+        return genders[2]
+      case 3:
+        return genders[3]
+    }
+  } */
+  return npc.gender
+}
```

Comment:
This one's weird. `profession.domSub` can be either ``"dom"` or `"sub"`, but the gender array values contains only numbers, `"male"`, and "female". So `genders.indexOf(profession.domSub)` will always return `-1`.

Reply:
This is a documentation issue- sorry. 

Essentially, professions are categorised into dominant and submissive roles, ergo `profession.domSub`.

However, because it's possible for the society to be matriarchal, the dominant gender isn't always male, therefore `genderEqualityLikelihood`- my intent was to link them up, but clearly forgot to do that...

Reply:
The function was running, it just wasn't ever returning the switch cases.

Reply:
Yup. Have rebuilt it to be a little clearer, and actually run; no more messing around, now it's just a straight "patriarchal or matriarchal" choice, with the value of `town.equality` actually being the equality; 0 is full sexism, 100 is equality of the sexes. Much simpler!